### PR TITLE
SWC-5723 Components to manage personal access tokens

### PR DESCRIPTION
**Pull request by nickgros**

### CODE REVIEW

File: src/lib/containers/CopyToClipboardInput.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
+import { library } from '@fortawesome/fontawesome-svg-core'
+import { faTimes } from '@fortawesome/free-solid-svg-icons'
+import * as React from 'react'
+import IconCopy from '../../lib/assets/icons/IconCopy'
+import { CSSTransition, TransitionGroup } from 'react-transition-group'
+
+library.add(faTimes)
```
Comment (nickgros) [inline src/lib/containers/CopyToClipboardInput.tsx:7]: remove fontawesome

File: src/lib/containers/CopyToClipboardInput.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    <React.Fragment>
+      <TransitionGroup>
+        {showModal && (
+          <CSSTransition
+            key={value}
+            classNames="SRC-card"
+            timeout={{ enter: 500, exit: 300 }}
+          >
+            <div key={value} className="SRC-modal">
+              Token copied to clipboard
```
Comment (nickgros) [inline src/lib/containers/CopyToClipboardInput.tsx:59]: Token -> prop

Comment (nickgros) [reply src/lib/containers/CopyToClipboardInput.tsx:59]: or just e.g. "Copied to clipboard"

File: src/lib/containers/CopyToClipboardInput.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            </div>
+          </CSSTransition>
+        )}
+      </TransitionGroup>{' '}
+      <input
+        className="form-control SRC-marginBottomTop"
+        type="text"
+        style={{
+          display: 'inline',
```
Comment (nickgros) [inline src/lib/containers/CopyToClipboardInput.tsx:68]: pull out to scss file

File: src/lib/containers/CopyToClipboardInput.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+          width: inputWidth,
+          marginRight: '5px',
+        }}
+        value={value}
+        readOnly={true}
+        onClick={copyToClipboard(value)}
+      ></input>
+      <div
+        ref={htmlDivRef}
+        style={{ display: 'inline' }}
```
Comment (nickgros) [inline src/lib/containers/CopyToClipboardInput.tsx:78]: pull out to scss

File: src/lib/containers/CopyToClipboardInput.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        style={{
+          display: 'inline',
+          width: inputWidth,
+          marginRight: '5px',
+        }}
+        value={value}
+        readOnly={true}
+        onClick={copyToClipboard(value)}
+      ></input>
+      <div
```
Comment (nickgros) [inline src/lib/containers/CopyToClipboardInput.tsx:76]: consider putting the icon in a button to indicate interactivity

Comment (nickgros) [reply src/lib/containers/CopyToClipboardInput.tsx:76]: or scss hover functionality

File: src/lib/utils/synapseTypes/AccessToken/ScopeDescriptions.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
+export const scopeDescriptions = {
```
[the remark this answers is not in the document]
Comment (nickgros) [reply src/lib/utils/synapseTypes/AccessToken/ScopeDescriptions.ts:1]: https://sagebionetworks.jira.com/browse/PLFM-6503

File: src/lib/index.ts [hunk 7]
```diff
@@ -70,6 +71,7 @@ const SynapseComponents = {
   UserCardListRotate,
   FeaturedDataTabs,
   UserCardListGroups,
+  AccessTokenPage,
```
Comment (nickgros) [inline src/lib/index.ts:74]: Remove

File: src/lib/containers/personal_access_token/AccessTokenPage.tsx [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+import { CreateAccessTokenModal } from './CreateAccessTokenModal'
+import { ErrorBoundary, FallbackProps } from 'react-error-boundary'
+
+const ErrorFallback: React.FunctionComponent<FallbackProps> = ({
+  error,
+  resetErrorBoundary,
+}) => {
+  return (
+    <div role="alert" className="SRC-marginBottomTop">
+      <p>The following error occurred:</p>
```
Comment (nickgros) [inline src/lib/containers/personal_access_token/AccessTokenPage.tsx:17]: Remove this

**Review by jay-hodgson — APPROVED** [review]

**Pull request merged**